qrenderdoc/Windows/BufferViewer.cpp

```diff
@@ -6638,6 +6682,122 @@ void BufferViewer::debugVertex()
   m_Ctx.AddDockWindow(s->Widget(), DockReference::AddTo, this);
 }
 
+void BufferViewer::debugMeshletThread()
+{
+  if(!m_Ctx.IsCaptureLoaded())
+    return;
+
+  const ActionDescription *action = m_Ctx.CurAction();
+  if(!action)
+    return;
+
+  if(!m_CurView)
+    return;
+
+  QModelIndex idx = m_CurView->selectionModel()->currentIndex();
+
+  if(!idx.isValid())
+  {
+    GUIInvoke::call(this, [this]() {
+      RDDialog::critical(this, tr("Error debugging"),
+                         tr("Error debugging meshlet - make sure a valid meshlet is selected"));
+    });
+    return;
+  }
+
+  uint32_t taskIndex = 0, meshletIndex = 0;
+  GetIndicesForMeshRow((uint32_t)idx.row(), taskIndex, meshletIndex);
+
+  uint32_t vertid =
+      m_CurView->model()->data(m_CurView->model()->index(idx.row(), 1), Qt::DisplayRole).toUInt();
+  uint32_t index =
+      m_CurView->model()->data(m_CurView->model()->index(idx.row(), 2), Qt::DisplayRole).toUInt();
+
+  const ShaderReflection *shaderDetails =
+      m_Ctx.CurPipelineState().GetShaderReflection(ShaderStage::Mesh);
+
+  if(!shaderDetails)
+    return;
+
+  m_MeshletDebugSelector->SetThreadBounds(action->dispatchDimension,
+                                          (action->dispatchThreadsDimension[0] == 0)
+                                              ? shaderDetails->dispatchThreadsDimension
+                                              : action->dispatchThreadsDimension);
+  m_MeshletDebugSelector->SetDefaultDispatch({meshletIndex, 0, 0}, {vertid, 0, 0});
+
+  RDDialog::show(m_MeshletDebugSelector);
+}
+
+void BufferViewer::meshletDebugSelector_beginDebug(const rdcfixedarray<uint32_t, 3> &group,
+                                                   const rdcfixedarray<uint32_t, 3> &thread)
+{
+  const ActionDescription *action = m_Ctx.CurAction();
+
+  if(!action)
+    return;
+
+  const ShaderReflection *shaderDetails =
+      m_Ctx.CurPipelineState().GetShaderReflection(ShaderStage::Mesh);
+
+  if(!shaderDetails)
+    return;
+
+  struct threadSelect
+  {
+    rdcfixedarray<uint32_t, 3> g;
+    rdcfixedarray<uint32_t, 3> t;
+  } debugThread = {
+      // g[]
+      {group[0], group[1], group[2]},
+      // t[]
+      {thread[0], thread[1], thread[2]},
+  };
+
+  bool done = false;
+  ShaderDebugTrace *trace = NULL;
+
+  m_Ctx.Replay().AsyncInvoke([&trace, &done, debugThread](IReplayController *r) {
+    trace = r->DebugMeshletThread(debugThread.g, debugThread.t);
+
+    if(trace->debugger == NULL)
+    {
+      r->FreeTrace(trace);
+      trace = NULL;
+    }
+
+    done = true;
+  });
+
+  QString debugContext = lit("Group [%1,%2,%3] Thread [%4,%5,%6]")
```

Comment:
Might be nice here to add `Mesh` to the front of this context string so the panel title is easier to see at a glance if someone is debugging both a compute shader and a mesh shader.